src/plugins/intel_cpu/src/nodes/kernels/stack_allocator.hpp

```diff
@@ -0,0 +1,374 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <string>
+#include <vector>
+#include <math.h>
+#include <dnnl_extension_utils.h>
+#include <cpu/x64/jit_generator.hpp>
+#include <stack>
+
+namespace ov {
+namespace intel_cpu {
+
+using namespace dnnl::impl;
+namespace x64 = dnnl::impl::cpu::x64;
+
+class StackAllocator final {
+public:
+    using Ptr = std::shared_ptr<StackAllocator>;
+
+    class Address;
+    template<typename TReg>
+    class Reg;
+
+    StackAllocator(x64::jit_generator& code_gen)
+        : StackAllocator{code_gen, code_gen.rbp} {
+    }
+
+    StackAllocator(x64::jit_generator& code_gen, const Xbyak::Reg& bp)
+        : StackAllocator{code_gen, bp, 1} {
+    }
+
+    StackAllocator(x64::jit_generator& code_gen,
+                   const Xbyak::Reg& bp,
+                   const size_t alignment)
```

Comment:
When do we need the stack aligned > 1?

Reply:
Yes, for example some sse4.1 commands requires address to be aligned

Reply:
@ceciliapeng2011 Do you have some questions related to alignment ?
If no, please resolve conversation